### DISCUSSION TITLE
Discord bot repeats itself and spams HTML code when requesting website data.

### DISCUSSION THREAD

**Discussion by sirquackiii** [Rapptz/discord.py#6546] [Q&A]
So I am making a discord bot with a command to do HTTP requests via beautifulsoup. Right now it's supposed to get the first `<h1>` tag and prints it in bold, and it will then get the first `<p>` tag and print it as well. In the code, I ask for it to get the __text__ of the tags specifically. But when I ask it to get the text from wikipedia, it repeats itself, prints out HTML code, and just goes nuts.

![molething](https://user-images.githubusercontent.com/77940766/111849912-07c52a80-88cc-11eb-94d0-38c4ceffa66d.png)

I tried the **exact same code** by just printing it to a terminal and it works perfectly fine.

Here is my code:
```
@commands.command()
async def web(ctx, site):
    r = requests.get(f"https://{site}")
    soup = BeautifulSoup(r.text, "lxml")
    h = soup.h1.text
    p = soup.p.text
    
    await ctx.send(h)
    await ctx.send(p)

    return
```
I am not sure what's going on here

**Comment by Rapptz** [comment]
This is for the discord.py library, not `BeautifulSoup`. The only advice I can offer here is that [`requests` is blocking](https://discordpy.readthedocs.io/en/latest/faq.html#what-does-blocking-mean) and shouldn't be used. Use `aiohttp` instead.

Other than that, this seems to do with external libraries rather than discord.py -- so it's off-topic here.
